lib/core/auth/scram.js

```diff
@@ -29,8 +29,8 @@ var parsePayload = function(payload) {
 };
 
 var passwordDigest = function(username, password) {
-  if (typeof username !== 'string') throw new MongoError('username must be a string');
-  if (typeof password !== 'string') throw new MongoError('password must be a string');
+  if (typeof username !== 'string') throw new MongoError('username must be a string, got: "' + username + '"');
+  if (typeof password !== 'string') throw new MongoError('password must be a string, got: "' + password + '"');
```

Comment:
Is there the possibility of logging the password in plain-text here? For example, if I do the following:

```js
...
password: new String('mypassword'),
...
```

then it will throw an error with the message: `password must be a string, got: "mypassword"`. I believe the same could occur with setting password to `{ toString() { return 'mypassword' } }`, though that seems less likely to happen in the real world.

Also wondering if this is an issue, since hopefully this error is not occurring on production boxes.

Reply:
That is a good point, I hadn't thought of that. Perhaps we instead print `typeof password` to avoid that possibility? The case I'm looking at is someone using `pass` instead of `password` and getting this error

Reply:
yeah, I think printing the `typeof` would definitely work here. In the case of using `pass` instead of `password`, it would wind up being `undefined` anyway.

Reply:
Done :+1: